pkg/cli/create_helm.go

```diff
@@ -148,17 +150,40 @@ func CreateHelm(ctx context.Context, options *CreateOptions, globalFlags *flags.
 		}
 	}
 
-	// TODO Delete after vCluster 0.19.x resp. the old config format is out of support.
-	if isVClusterDeployed(release) {
-		migratedValues, err := migrateLegacyHelmValues(release)
-		if err != nil {
-			return err
+	// TODO Refactor after vCluster 0.19.x resp. the old config format is out of support.
+	// Early abort if a user runs a virtual cluster < v0.20 without providing any values files during an upgrade.
+	// We do this because we don't want to "automagically" convert the old config implicitly, without the user
+	// realizing that the virtual cluster is running with the old config format.
+	if isVClusterDeployed(release) && isLegacyVCluster(release.Chart.Metadata.Version) && len(cmd.Values) <= 0 {
+		// If we have a < v0.20 virtual cluster running we have to infer the distro from the current chart name.
+		currentDistro := strings.TrimPrefix(release.Chart.Metadata.Name, "vcluster-")
+		// If we are upgrading a vCluster < v0.20 the old k3s chart is the one without a prefix.
+		if currentDistro == "vcluster" {
+			currentDistro = config.K3SDistro
 		}
 
-		// we do not simply upgrade a vcluster without providing a values.yaml file when it is running with a previous release, as this might result in a distro/backingstore switch.
-		if len(cmd.Values) <= 0 && migratedValues != "" {
-			return fmt.Errorf("abort upgrade: no values files were provided while the current virtual cluster is running with an old values format (< v0.20)\nConsider updating your config values to the following new v0.20 format:\n\n%s", migratedValues)
+		// A virtual cluster could either be created via vcluster CLI or via helm.
+		// When using vcluster CLI we always have extra values configured.
+		// When using helm without any modifications we don't have any extra values,
+		// so we must take the default values from the release into account.
+		var currentValues []byte
+		if release.Config != nil {
+			currentValues, err = yaml.Marshal(release.Config)
+			if err != nil {
+				return err
+			}
+		} else {
+			currentValues, err = helmValuesYAML(release)
+			if err != nil {
+				return err
+			}
 		}
+
+		command := fmt.Sprintf("vcluster convert config --distro %s - <<EOF\n%s\nEOF", currentDistro, string(currentValues))
```

Comment:
This will print a big chunk, can we not do something like:
```
helm get values ... | vcluster convert config ...
```

Reply:
Sure. Just to be sure. `helm get values` will retrieve the latest revision right? I tested locally and it looks like it 🤔